base/number.jl

```diff
@@ -63,4 +63,6 @@ zero{T<:Number}(::Type{T}) = convert(T,0)
 one(x::Number)  = oftype(x,1)
 one{T<:Number}(::Type{T}) = convert(T,1)
 
+_default_type(::Type{Number}) = Int
```

Comment:
I would prefer to leave it just as `Number`.  Having `Base.promote_op(+, Int, Number) == Number`  seems fine to me.

Moreover, this seems unrelated to #17984 and #18176 ... in general, @pabloferz, it makes it much harder to review PRs if you include unrelated changes in the PR.